### DISCUSSION TITLE
How to export a file showing the differences as highlighted in Winmerge?

### DISCUSSION THREAD

**Discussion by kelleycan2** [WinMerge/winmerge#920] [Q&A]
Winmerge does such a beautiful job of comparing files and allowing the use of colors to easily see the differences between 2 files!  

But once you have viewed those differences in Winmerge, how do you get that data OUT of Winmerge so you can do something with it in another program such as a spreadhseet?

Is there an "export differences" function I'm not finding?

Best I've been able to do so far is print to PDF and then use Adobe tools convert the PDF to an Excel spreadsheet.  Very ugly results and requires tons of manual work in spreadsheet to make it usable.  Any other ways to do this???

Thanks for any input or assistance!
Kelley

**Comment by sdottaka** [comment]
WinMerge can create an HTML File Compare Report from the Tools-> Generate Report menu item.

![image](https://user-images.githubusercontent.com/98126/129429710-e8458984-b1a8-4a5a-9f46-a565c253827a.png)

![image](https://user-images.githubusercontent.com/98126/129429729-c933a8ad-e5e3-4a23-8260-47efdc39facf.png)

**Reply by AdrienViala** [reply]
nice feature, simple and efficient. love it

**Comment by kelleycan2** [comment]
Thank you!  That is very helpful.  And the good news is that copy/pasting
the html "report" into a spreadsheet DOES retain the formatting and
side-by-side comparison results.  Perfect!

Kelley Canon
Language & Culture Archives
REAP Administrator
SIL International

**Comment by illtellyoulater** [comment]
I'm trying to only export different lines. Is there a way to do that?
I've also tried the provided plugins but couldn't make them work.

Update: found the solution myself: https://github.com/WinMerge/winmerge/discussions/1226 (very clever!)

**Comment by native-api-work** [comment]
Unfortunately, "Generate report" ignores the mode of comparison.
I'm comparing in Table mode, but the report regenerates the result of a comparison in Text mode.
That's significant because in text mode, some line correspondences are detected incorrectly.

**Comment by nokeeffe-dil** [comment]
Can you only export differences on a file level? I have a number of changed files in different folders, so I would like to run a comparison and export all of these changes to one file. There are far too many files to go file-by-file, so looking for a solution to export the differences from a folder level. Not just a list of the files that are different, but the differences _**within**_ each file. Thanks.

**Reply by sdottaka** [reply]
Select the files you want to output the differences for, and select Tools → Generate Patch menu to output the patch. The differences between multiple files will be output to a single file.

![image](https://github.com/user-attachments/assets/2132c57e-390b-4fe6-a0a3-359e9d16df13)
![image](https://github.com/user-attachments/assets/cc9df5f6-6094-47fd-b210-a58228b244ea)

**Comment by bravilordesign** [comment]
Windows:
1. Change the diff lines to "0 lines" (if you only want the differences, otherwise skip)
2. Ctrl + P for print dialog
3. Select "Microsoft print to PDF"
4. Select a file name and destination

You got your differences in a PDF file, which you can easily share with your team/stakeholders/whoever
